bigchaindb/models.py

```diff
@@ -376,6 +392,28 @@ def decouple_assets(self):
 
         return (assets, block_dict)
 
+    def decouple_metadata(self, block_dict=None):
+        """
+        Extracts the metadata from transactions in the block.
+
+        Returns:
+            tuple: (metadatas, block) with the metadatas being a list of dict/null and
+            the block being the dict of the block with no metadata in any transaction.
+        """
+        if block_dict is None:
+            block_dict = deepcopy(self.to_dict())
+
+        metadatas = []
+        for transaction in block_dict['block']['transactions']:
+            metadata = transaction.pop('metadata')
+            if isinstance(metadata, dict):
```

Comment:
`metadata` is either `None` or a `dict`. This is enforced by the schema validation. So a check to see if it's not None should be enough, right?